python/astro_metadata_translator/translators/megaprime.py

```diff
@@ -183,3 +185,63 @@ def to_observation_counter(self):
             The observation counter.
         """
         return self.to_exposure_id()
+
+    @classmethod
+    def determine_translatable_headers(cls, filename, primary=None):
+        """Given a file return all the headers usable for metadata translation.
+
+        MegaPrime files are multi-extension FITS with a primary header and
+        each detector stored in a subsequent extension.  MegaPrime uses
+        ``INHERIT=F`` therefore the primary header will always be ignored
+        if given.
```

Comment:
Does "ignored" mean "not returned in `headers`"?

Reply:
It means that whatever you supply as `primary` will make no difference to what is returned. INHERIT=F declares that each HDU has a standalone header that depends on no other header.

Reply:
Then, as with the DECam translator, I suggest explicitly saying that the returned headers don't include the primary header.

Reply:
To me "ignored" implies "will not be used". I have explicitly stated this now also in the yield line.